gdrapi.c

```diff
@@ -350,9 +350,15 @@ int gdr_copy_to_bar(void *bar_ptr, const void *h_ptr, size_t size)
             memcpy_uncached_store_sse(bar_ptr, h_ptr, size);
             break;
         }
+
         // fall through
         gdr_dbgc(1, "using plain implementation of gdr_copy_to_bar\n");
         memcpy(bar_ptr, h_ptr, size);
+
+        // fencing is needed even for plain memcpy(), due to performance
+        // being hit by delayed flushing of WC buffers
+        _mm_sfence();
```

Comment:
Not sure if the instruction actually flushes or if it just prevents re-ordering. The description [here](https://msdn.microsoft.com/es-cr/library/5h2w73d1(v=vs.90).aspx) doesn't indicate so. 
For performance to have improved may be the fence has a flush side-effect? Lmk your thoughts

Reply:
Sfence is known to flush the CPU WC buffer. Not easy to find a doc ref though.